secure/json-rpc/src/lib.rs

```diff
@@ -177,13 +177,22 @@ impl JsonRpcClient {
     }
 
     // Executes the specified request method using the given parameters by contacting the JSON RPC
-    // server.
+    // server. If the 'https_proxy' environment variable is set, enable the proxy.
     fn execute_request(&self, method: String, params: Vec<Value>) -> Response {
-        ureq::post(&self.host)
-            .timeout_connect(REQUEST_TIMEOUT)
-            .send_json(
-                json!({"jsonrpc": JSON_RPC_VERSION, "method": method, "params": params, "id": 0}),
-            )
+        let mut request = ureq::post(&self.host)
+            .timeout_connect(REQUEST_TIMEOUT).build();
+
+        if let Ok(proxy) = env::var("https_proxy") {
```

Comment:
Technically this should use `http_proxy` if the URL is HTTP vs HTTPS, but I think this is fine for now.

Reply:
Aah. thanks! I've updated the code to support both. What do you think?

Reply:
Perfect, thanks!